### DISCUSSION TITLE
Infrastructure Community [General Information]

### DISCUSSION THREAD

**Discussion by antonkri** [eclipse-score/score#107] [Infrastructure Community]
# Infrastructure Community

The Infrastructure Community designs and operates the shared systems and services that enable SCORE development and integration.  
We provide a common foundation so that teams can focus on building features, while relying on stable infrastructure and tooling.

**Vision:**  
We aim to apply modern software engineering practices wherever the SCORE process allows.

**Mission:**  
- Provide reliable and transparent infrastructure for all SCORE developers.  
- Align infrastructure and tooling across teams, avoiding fragmented solutions.  
- Empower new contributors by lowering the entry barrier through good defaults and documentation.  
- Ensure process compliance.

--- 

## Responsibilities

We deliver in four main areas:

- **Infrastructure**
  - Repositories follow consistent structure.
  - CI/CD pipelines are reliable, reproducible, and fast.
  - Integration follows modern approaches.

- **Toolchains**
  - Common toolchains for C++, Rust, and Python are available and maintained.  
  - Tool decisions for C++ and Rust are coordinated with Architecture/Process communities when they affect development directly.

- **Tools**
  - Internal tools improve developer productivity and reduce manual effort.  
  - External tools are integrated smoothly into the SCORE environment.

- **Docs-as-Code**
  - Documentation-as-code tooling is adjusted to our processes.  

---

## Members

There is no strict team assignment — contributions are open to everyone.  
Repository maintainers are defined in each repo’s `CODEOWNERS` file.

- Community Lead: @AlexanderLanin 
- Proxy: @MaximilianSoerenPollak 

---

## Meetings

* Monday 15:00 CEST (UTC+2) - Infrastructure Weekly
* Wednesday 13:00 CEST (UTC+2) - Rust Meeting (making the toolchain work)

- [iCal (Outlook import)](https://calendar.google.com/calendar/ical/c_2ampi2bmoka3qter4dceap1d5g%40group.calendar.google.com/public/basic.ics)  
- [Web View](https://calendar.google.com/calendar/u/0/embed?src=c_2ampi2bmoka3qter4dceap1d5g@group.calendar.google.com&ctz=Europe/Berlin)  
- [Google Calendar](https://calendar.google.com/calendar/u/0?cid=Y18yYW1waTJibW9rM3F0ZXI0ZGNlYXAxZDUnZ0Bncm91cC5jYWxlbmRhci5nb29nbGUuY29t)
- [Meeting Minutes](https://github.com/orgs/eclipse-score/discussions/236)

---

## Slack Channel
https://sdvworkinggroup.slack.com/archives/C0894QGRZDM

---

## Repositories

https://github.com/eclipse-score#infrastructure

---

## Learning Resources

Want to contribute, but not sure where to start?  
Here are some topics that will help (links to be added later):

- Bazel  
- Python 3.12 / 3.13  
- Modern Software Engineering
  - trunk based development
- GitHub Workflows  
- ISO 26262 Part 6 / SCORE process  

---

## How to Contribute

- **Issues:** Create issues in the `score` repo.  
  Add the label `community:infrastructure` to have them tracked in our [GitHub project board](https://github.com/orgs/eclipse-score/projects/6).  

- **Pull Requests:** Follow SCORE contribution guidelines; maintainers will review.  

- **Join Meetings:** See links above for schedules and access.

**Comment by EGanesh080898** [comment]
Issue

I am trying to build SCORE communication module for QNX 8.0 using:

bazel build --config=com-x86_64-qnx //score/mw/com/example/ipc_bridge:ipc_bridge_cpp

But the build fails with:

glob pattern 'host/linux/x86_64/**/*' didn't match anything

My local SDP path is:

/home/qnx800

And inside it I have:

host/linux/x86_64/etc
host/linux/x86_64/usr

bazel_dep(name = "score_toolchains_qnx", version = "0.0.2", dev_dependency=True)
toolchains_qnx = use_extension("@score_toolchains_qnx//:extensions.bzl", "toolchains_qnx", dev_dependency=True)
toolchains_qnx.sdp(
url = "file:///home/qnx800.tar.gz",
sha256 = "d8e65444ad9a3da91187dc5898447a84ec4f60d2e46b0eacdb8cde21834ebadc",
)
use_repo(toolchains_qnx, "toolchains_qnx_sdp")
use_repo(toolchains_qnx, "toolchains_qnx_qcc")

But Bazel seems to expect more directories under host/linux/x86_64.

What I need help with

Is the pre-packaged QNX 8.0 SDP supported by SCORE?

Does SCORE require a full SDP installation?

What is the correct expected folder structure for host/linux/x86_64?

How can I fix this glob error?

Thank you!

**Reply by EGanesh080898** [reply]
@antonkri @AlexanderLanin  can you please look into my issue and help me out.
